mlir/lib/Conversion/PDLToPDLInterp/PredicateTree.cpp

```diff
@@ -875,6 +885,27 @@ static void insertExitNode(std::unique_ptr<MatcherNode> *root) {
   *root = std::make_unique<ExitNode>();
 }
 
+/// Sorts the range begin/end with the partial order given by cmp.
+template <typename Iterator, typename Compare>
+void stableTopologicalSort(Iterator begin, Iterator end, Compare cmp) {
```

Comment:
this method should be static. Also, does it make sense to put somewhere else, like STLExtras?

Reply:
@Mogball, I made the function static. Is that fine with you?

Reply:
static is fine